pkg/sharing/secret_exports.go

```diff
@@ -93,14 +96,14 @@ type SecretMatcher struct {
 //       - secret with specific namespace
 //       - secret with wildcard namespace match
 //     (in all cases fallback to secret namespace/name sort)
-func (se *SecretExports) MatchedSecretsForImport(matcher SecretMatcher) []*corev1.Secret {
+func (se *SecretExports) MatchedSecretsForImport(matcher SecretMatcher, nsIsExcluded NamespaceExclusionCheck) []*corev1.Secret {
```

Comment:
im also thinking of may be we should change "behaviour/naming" -- DoesNamespaceAcceptSecrets() bool

Reply:
re. putting it in the struct vs. passing as an arg -  i went back and forth and obviously settled here.  If the `ToNamespace` is already known when we make the struct there's an argument of like, "why didn't you invoke the function on the name, and then not bother making the struct at all if the check was going to fail?"


i don't think `DoesNamespaceAcceptSecrets` works,  because per your spec it _does_ accept secrets if it's explicitly named.   so it's back to `DoesNamespaceAcceptWildcardSecretExports` or something similarly out of some Java architect's BufferedReaderWriter fantasy. 

... i haven't come up with a decent short name.  not trying to defend whatever i used yesterday,  but not totally sold on the longer versions of `NsSezYesToStarShares`